crates/router/src/core/payments.rs

```diff
@@ -3451,10 +3468,45 @@ pub async fn get_session_token_for_click_to_pay(
             merchant_country_code: click_to_pay_metadata.merchant_country_code,
             transaction_amount,
             transaction_currency_code: transaction_currency,
+            phone_number: optional_customer_details
+                .as_ref()
+                .and_then(|details| details.phone.clone().expose_option()),
+            email: optional_customer_details
+                .as_ref()
+                .and_then(|details| details.email.clone()),
+            phone_country_code: optional_customer_details
+                .and_then(|details| details.phone_country_code.clone()),
         },
     )))
 }
 
+fn validate_customer_details_for_click_to_pay(customer_details: &CustomerData) -> RouterResult<()> {
+    if customer_details.email.is_some() {
+        return Ok(());
+    }
+
+    match (
+        customer_details.phone.as_ref(),
+        customer_details.phone_country_code.as_ref(),
+    ) {
+        (Some(_), Some(_)) => Ok(()),
+        (None, None) => Err(errors::ApiErrorResponse::MissingRequiredFields {
```

Comment:
Ideally we should be mentioning in the error that, either email or phone number is required for ctp flow. So, that merchant would do the integration accordingly right?

Reply:
sure, will add another level of error check